### PR TITLE
[Admin REST API]- Role-Alias mapping

### PR DESCRIPTION
**Pull request by Meruja**
This PR contains GET and PUT operations for role-alias mapping. The original role can be mapped with a different name when customizing. 
GET :
Request : curl  -k -X GET 'https://localhost:9443/api/am/admin/v1/system-scopes/role-aliases' -H "Authorization: Bearer eb435e9d-f067-38b3-901a-c1104ca0095d" 
Response : `{
    "count": 2,
    "list": [
        {
            "role": "Internal/subscriber",
            "aliases": [
                "testing,test"
            ]
        },
        {
            "role": "Internal/creator",
            "aliases": [
                "alias"
            ]
        }
    ]
}`

PUT :
Request  : curl -k -X PUT 'https://localhost:9443/api/am/admin/v1/system-scopes/role-aliases' -H 'Authorization: Bearer 2992c6bc-08e7-3051-ace9-be2ce3111392'  -H 'Content-Type: application/json' 
-d  '{"count":1,"list":[{ "role":"Internal/subscriber","aliases":"testing,test"}, {"role": "Internal/creator","aliases": "alias"}]}'
Response: `{
    "count": 2,
    "list": [
        {
            "role": "Internal/subscriber",
            "aliases": [
                "testing,test"
            ]
        },
        {
            "role": "Internal/creator",
            "aliases": [
                "alias"
            ]
        }
    ]
}`

Related to https://github.com/wso2/product-apim/issues/7954

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/impl/SystemScopesApiServiceImpl.java [hunk 1]
```diff
@@ -33,33 +44,34 @@
...
-                }
+        boolean existence;
+
+        if (username == null) {
+            existence = apiAdmin.isScopeExists(RestApiUtil.getLoggedInUsername(), decodedScope);
+            if (existence) {
+                scopeSettingsDTO.setName(decodedScope);
+            } else {
+                throw new APIManagementException("Scope Not Found. Scope : " + decodedScope +
+                        ExceptionCodes.SCOPE_NOT_FOUND);
```
Comment (rmsamitha) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/impl/SystemScopesApiServiceImpl.java:60]: ExceptionCodes.SCOPE_NOT_FOUND should be a separate argument to the Constructor

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/impl/SystemScopesApiServiceImpl.java [hunk 2]
```diff
@@ -71,4 +83,59 @@ public Response systemScopesGet(MessageContext messageContext) throws APIManagem
...
+    }
+
+    @Override
+    public Response systemScopesRoleAliasesPut(RoleAliasListDTO body, MessageContext messageContext)
+            throws APIManagementException {
+        String tenantDomain = MultitenantUtils.getTenantDomain(RestApiUtil.getLoggedInUsername());
+        RoleAliasListDTO roleAliasListDTO = new RoleAliasListDTO();
+        //read from tenant-conf.json
+        JsonObject existingTenantConfObject = new JsonObject();
+
```
Comment (rmsamitha) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/impl/SystemScopesApiServiceImpl.java:107]: Remove empty line

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/resources/admin-api.yaml [hunk 3]
```diff
@@ -3535,6 +3535,81 @@ paths:
...
+  # The 'Role Alias' Collection API Resource
+  ######################################################
+  /system-scopes/role-aliases:
+    #-----------------------------------------------------
+    # Retrieve role alias mappings for system scope roles
+    #-----------------------------------------------------
+    get:
+      summary: Retreive role alias mappings
+      operationId: systemScopesRoleAliasesGet
+      security:
```
Comment (rmsamitha) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/resources/admin-api.yaml:3548]: x-wso2-curl, x-wso2-request, x-wso2-response should be defined

Comment (malinthaprasan) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/resources/admin-api.yaml:3548]: @rmsamitha we need to use x-examples instead. Let's define it for now.

On a separate note, we need to rethink the way we define examples in REST APIs.

Comment (Meruja) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/resources/admin-api.yaml:3548]: As per https://github.com/wso2/product-apim/issues/8231, x-wso2-curl, x-wso2-request, and x-wso2-response will be added as x-examples.
[Example for x-examples.](https://github.com/wso2/carbon-apimgt/blob/master/components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml#L128)

Comment (rmsamitha) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/resources/admin-api.yaml:3548]: +1 for x-examples

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/impl/SystemScopesApiServiceImpl.java (hunk 1 above)
Comment (Meruja) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/impl/SystemScopesApiServiceImpl.java:60]: Fixed with 570298d

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/impl/SystemScopesApiServiceImpl.java (hunk 2 above)
Comment (Meruja) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/impl/SystemScopesApiServiceImpl.java]: Fixed with 570298d

**Pull request closed without merging**